cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java

```diff
@@ -500,6 +545,32 @@ public ClusterModel clusterModel(long from,
     return clusterModel;
   }
 
+  /**
+   * Populate replica placement information, i.e. replica resides on which disk of the broker, to the cluster model.
+   *
+   * @param clusterModel The cluster model to populate replica placement information.
+   */
+  private void populateReplicaPlacementInfoForClusterModel(ClusterModel clusterModel) {
+    Map<Integer, KafkaFuture<Map<String, LogDirInfo>>> logDirsByBrokerId =
+        _adminClient.describeLogDirs(clusterModel.aliveBrokers().stream().mapToInt(Broker::id).boxed().collect(Collectors.toList())).values();
+    for (Map.Entry<Integer, KafkaFuture<Map<String, LogDirInfo>>> entry : logDirsByBrokerId.entrySet()) {
+      Broker broker = clusterModel.broker(entry.getKey());
+      try {
+        entry.getValue().get(LOGDIR_RESPONSE_TIMEOUT_MS, TimeUnit.MILLISECONDS).forEach((key, value) -> {
+          if (value.error == Errors.NONE) {
+            broker.populateDiskInfo(key, value.replicaInfos);
+          } else {
+            broker.disk(key).setState(DEAD);
+          }
+        });
+      } catch (TimeoutException te) {
+        LOG.error("Getting logdir information for broker {} timed out.", entry.getKey());
+      } catch (InterruptedException | ExecutionException e) {
+        LOG.error("Populating logdir information for broker {} encountered Exception {}.", entry.getKey(), e);
```

Comment:
As discussed offline, in case of any exception that prevents CC retrieve the disk capacities, the right action should be to throw an exception to indicate issues with Kafka -- i.e. we should not quietly work on incomplete information.